saleor/graphql/product/tests/test_product.py

```diff
@@ -950,6 +950,21 @@ def test_products_query_with_filter_stocks(
     assert {node["node"]["id"] for node in products_data} == product_ids
 
 
+def test_query_products_query_with_filter_ids(
+    staff_api_client, product, query_products_with_filter
+):
+    product_global_id = graphene.Node.to_global_id("Product", product.id)
+    variables = {"filter": {"ids": [product_global_id]}}
+    response = staff_api_client.post_graphql(
+        query_products_with_filter, variables, check_no_permissions=False
```

Comment:
`check_no_permissions` has no effect if you didn't pass `permissions`.
```suggestion
        query_products_with_filter, variables
```